PodcastGenerator/core/episodes.php

```diff
@@ -92,6 +97,70 @@ function getEpisodes($category = null, $_config)
     return $episodes_data;
 }
 
+function searchEpisodes($name = "", $_config)
+{
+    $name = strtolower($name);
+    $episodes_mtimes = setupEpisodes($_config);
+    // Check if name is a category and replace
+    $cats_xml = simplexml_load_file('categories.xml');
+    foreach ($cats_xml as $item) {
+        if ($name === strtolower($item->description[0])) {
+            $name = strval($item->id[0]);
+        }
+    }
+    // Get XML data for the episodes of interest.
+    $episodes_data = array();
+    for ($i = 0; $i < sizeof($episodes_mtimes); $i++) {
+        $episode = $episodes_mtimes[$i][0];
+        // We need to get the CDATA in plaintext.
+        $xml_file_name = pathinfo($_config['absoluteurl'] . $_config['upload_dir'] . $episode, PATHINFO_FILENAME) . '.xml';
+        $xml = simplexml_load_file($_config['absoluteurl'] . $_config['upload_dir'] . $xml_file_name, null, LIBXML_NOCDATA);
+        foreach ($xml as $item) {
+            if (strpos(strtolower($item->titlePG), $name) === false
+                && strpos(strtolower($item->shortdescPG[0]), $name) === false
+                && strpos(strtolower($item->longdescPG[0]), $name) === false
+                && strpos(strtolower($item->categoriesPG->category1PG[0]), $name) === false
+                && strpos(strtolower($item->categoriesPG->category2PG[0]), $name) === false
+                && strpos(strtolower($item->categoriesPG->category3PG[0]), $name) === false
+                && strpos(strtolower($item->keywordsPG[0]), $name) === false
+                && strpos(strtolower($item->authorPG->namePG[0]), $name) === false) {
+                continue;
+            }
+            $append_array = [
```

Comment:
As we use this piece of code two times (I think), can't we just create an internal function which takes $item as parameter and returns that array? We could save writing more code then

Reply:
Also thought of that, added and tested.